### PR TITLE
Add Two Trees Sapphire Pro Marlin Settings

### PR DESCRIPTION
**Pull request by ils15**
Sapphire Pro Marlin Example File with Color UI

Enjoy

### CODE REVIEW

File: config/examples/Two Trees/Sapphire Pro/Configuration.h [hunk 1]
```diff
@@ -0,0 +1,2501 @@
...
+ *
+ * :[-1, 0, 1, 2, 3, 4, 5, 6, 7]
+ */
+#define SERIAL_PORT 3
+
+/**
+ * Select a secondary serial port on the board to use for communication with the host.
+ * :[-1, 0, 1, 2, 3, 4, 5, 6, 7]
+ */
+#define SERIAL_PORT_2 -1
```
Comment (meteozond) [inline config/examples/Two Trees/Sapphire Pro/Configuration.h]: It fails to build with
```
In file included from Marlin/src/HAL/STM32F1/../../inc/MarlinConfig.h:49:0,
                 from Marlin/src/HAL/STM32F1/HAL.cpp:30:
Marlin/src/HAL/STM32F1/../../inc/SanityCheck.h:2310:6: error: #error "SERIAL_PORT_2 is set to -1, but the MOTHERBOARD has no native USB support. Set SERIAL_PORT_2 to a valid value for your board."
     #error "SERIAL_PORT_2 is set to -1, but the MOTHERBOARD has no native USB support. Set SERIAL_PORT_2 to a valid value for your board."
```

Comment (ils15) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: Oh sorry for that
please set serial port_2 1

i've ported it from my robin nano V2 board and forget to change this values
thanks

Comment (meteozond) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: Hm, is it should be fixed too?
https://github.com/MarlinFirmware/Configurations/pull/336/files/2d9018d4f8ea85c4387cddcd9ac685725a30a090#diff-a0ec2751395d0460bbd0723f27b06bea0707681b4985cdf2a2bb4cd48352942dR132

Comment (ils15) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: I believe only this line is incorrect
if you have another issue, please tell me

Comment (meteozond) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: It doesn't boot stacks at `TFT Updating 100%`

Comment (ils15) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: can be your sd card
original TT SD card is a crap
try to format it and update again

Comment (meteozond) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: I've seen in le3tspeak's fork that "robin_nano"
https://github.com/le3tspeak/Marlin-2.0.X-MKS-Robin-Nano/blob/MKS-Robin-Nano/Marlin/Configuration.h#L372

Comment (ils15) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: i use my own marlin in sapphire pro
here everything works fine
flash problem may be a sd card faulty

Comment (meteozond) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: I've used 2.0-bugfix and looking through configuration.h I've found some newer changes missing in you PR.

Comment (meteozond) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: Tried new SD card with same result.

Comment (ils15) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: Florian Marlin works?
did you use latest bugfix to compile config file?

Comment (meteozond) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: I don't know I've stated with your 😌

**Comment by ils15** [comment]
No problem
try it
[Robin_nano35.zip](https://github.com/MarlinFirmware/Configurations/files/5585664/Robin_nano35.zip)

File: config/examples/Two Trees/Sapphire Pro/Configuration.h (hunk 1 above)
Comment (ils15) [reply config/examples/Two Trees/Sapphire Pro/Configuration.h]: i solved problem
i dont know why my PR isnt update here
now everything will work

**Comment by meteozond** [comment]
Robin_nano35.zip flashed and started, trying your last version

**Comment by ils15** [comment]
> Robin_nano35.zip flashed and started, trying your last version

thanks for help and feedback
and sorry for all incovenience
i believe things will be work now
if you have some request, please talk to me as fast as you can

**Comment by meteozond** [comment]
Still no luck :(
Looks like BAUDRATE should be 115200

**Comment by ils15** [comment]
> Still no luck :(
> Looks like BAUDRATE should be 115200

no image?
print doesnt start?
baudrate isnt the problem

my compiled version works?
is the same file attached
please update your config.h file

**Comment by meteozond** [comment]
I've already dropped all my changes :( 
updated `Configuration.h` to 7a5a59a
It still stacks
image

IMG_1023

Your binary works fine
IMG_1024

**Comment by ils15** [comment]
Please download latest marlin bugfix and make your configs based on my file
use notepad++ and compare plugin or vscode compare function....
things changed a lot since florian firmware

**Comment by meteozond** [comment]
Just updated Marlin - rebuilding with your configs

**Comment by meteozond** [comment]
Checked out fresh bugfix-2.0.x 
Added your latest configs - still no result :(
Couldn't you build your Marlin bugfix-2.0.x version binary for me?
Just to be sure if it is my mistake.

Sorry for delay, I've broken my sd adapter 🤦‍♂️ It took some time to find working one

File: config/examples/Two Trees/Sapphire Pro/Configuration.h [hunk 2]
```diff
@@ -0,0 +1,2501 @@
...
+#define PID_MAX BANG_MAX // Limits current to nozzle while PID is active (see PID_FUNCTIONAL_RANGE below); 255=full current
+#define PID_K1 0.95      // Smoothing factor within any PID loop
+
+#if ENABLED(PIDTEMP)
+  #define PID_EDIT_MENU         // Add PID editing to the "Advanced Settings" menu. (~700 bytes of PROGMEM)
+  #define PID_AUTOTUNE_MENU     // Add PID auto-tuning to the "Advanced Settings" menu. (~250 bytes of PROGMEM)
+  //#define PID_PARAMS_PER_HOTEND // Uses separate PID parameters for each extruder (useful for mismatched extruders)
+                                  // Set/get with gcode: M301 E[extruder number, 0-2]
+
+  #define PID_EDIT_MENU         // Add PID editing to the "Advanced Settings" menu. (~700 bytes of PROGMEM)
```
Comment (meteozond) [inline config/examples/Two Trees/Sapphire Pro/Configuration.h]: Looks like some duplication is here

**Comment by meteozond** [comment]
I'm giving up. I don't know what the problem is. I've tried building with mac and windows machines, installing vscode+pio from scratch, older platform.io version - neither gave me sufficient result with any of existing Sapphire Pro Marlin forks :( Always stack at TFT Updating 100% and there is no any information on web.

**Comment by meteozond** [comment]
I've found the problem, remembering previous builds experience I was coping `.pio/build/mks_robin_nano35/firmware.bin` instead of `.pio/build/mks_robin_nano35/Robin_nano35.bin` 😡
So easy to mistake. 
Thanks a lot

File: config/examples/Two Trees/Sapphire Pro/Configuration.h [hunk 3]
```diff
@@ -0,0 +1,2495 @@
...
+
+/**
+ * TFT Rotation. Set to one of the following values:
+ *
+ *   TFT_ROTATE_90,  TFT_ROTATE_90_MIRROR_X,  TFT_ROTATE_90_MIRROR_Y,
+ *   TFT_ROTATE_180, TFT_ROTATE_180_MIRROR_X, TFT_ROTATE_180_MIRROR_Y,
+ *   TFT_ROTATE_270, TFT_ROTATE_270_MIRROR_X, TFT_ROTATE_270_MIRROR_Y,
+ *   TFT_MIRROR_X, TFT_MIRROR_Y, TFT_NO_ROTATION
+ */
+#define TFT_ROTATION TFT_NO_ROTATION
```
Comment (meteozond) [inline config/examples/Two Trees/Sapphire Pro/Configuration.h:2336]: This doesn't make sense. `TFT_NO_ROTATION` is default value

**Comment by sjasonsmith** [comment]
@ils15 What is the status of this? Are there issues remaining to be fixed, or is this ready to be reviewed?

**Comment by ils15** [comment]
> @ils15 What is the status of this? Are there issues remaining to be fixed, or is this ready to be reviewed?

everything works fine

**Comment by meteozond** [comment]
> > @ils15 What is the status of this? Are there issues remaining to be fixed, or is this ready to be reviewed?
> 
> everything works fine

confirm, everything working fine

**Comment by FlamingLasrSwrd** [comment]
It works for me as well.

File: config/examples/Two Trees/Sapphire Pro/Configuration.h [hunk 4]
```diff
@@ -0,0 +1,2513 @@
...
+// Ender-3 v2 OEM display. A DWIN display with Rotary Encoder.
+//
+//#define DWIN_CREALITY_LCD
+
+//
+// ADS7843/XPT2046 ADC Touchscreen such as ILI9341 2.8
+//
+#define TOUCH_SCREEN
+#if ENABLED(TOUCH_SCREEN)
+  #define BUTTON_DELAY_EDIT  50 // (ms) Button repeat delay for edit screens
```
Comment (meteozond) [inline config/examples/Two Trees/Sapphire Pro/Configuration.h:2352]: Just found that 50 to small for delay. Numbers change to maximum too fast, 75 is much more comfortabe.

**Pull request merged**